osf/models/base.py

```diff
@@ -506,129 +478,159 @@ def remove_guid_annotations(self):
             if table_name in self.query.tables:
                 del self.query.tables[self.query.tables.index(table_name)]
 
-    def _clone(self, annotate=False, **kwargs):
-        query = self.query.clone()
-        if self._sticky_filter:
-            query.filter_is_sticky = True
-        if annotate:
+    if not settings.CACHEOPS_ENABLED:
```

Comment:
Mind making this more like a pythonic preprocessor definition? e.g. https://stackoverflow.com/questions/482014/how-would-you-do-the-equivalent-of-preprocessor-directives-in-python

```python
if not settings.CACHEOPS_ENABLED:
    class GuidMixinQuerySet(object):

        def annotate_query_with_guids(self):
            return self

        # ...
else:
    class GuidMixinQuerySet(object):
        # ...
```